### PR TITLE
Complaint edit

### PR DESCRIPTION
**Pull request by manojkrsaini09**
added complaint edit functionlity and fixed some minor issues in create, detail

### CODE REVIEW

File: src/main/java/creativei/controller/ComplaintController.java [hunk 1]
```diff
@@ -27,30 +33,46 @@
...
             Page<Complaint> complaintList = complaintService.getAllComplaint(pageable);
             if(complaintList == null){
                 logger.error("No complaint found.");
-                return ResponseObject.getResponse(ExceptionType.NO_DATA_FOUND.getMessage()
-                        , ExceptionType.NO_DATA_FOUND.getCode());
+                return new ResponseEntity<>(ResponseObject.getResponse(ExceptionType.NO_DATA_FOUND.getMessage(),
+                        ExceptionType.NO_DATA_FOUND.getCode()), HttpStatus.NO_CONTENT);
             }
-            return ResponseObject.getResponse(complaintList);
+            return new ResponseEntity<>(ResponseObject.getResponse(ResponseHelper.getComplaintVOList(complaintList.getContent())),
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: You need to return Page<Complaint> object to use the power of paged object to apply pagination on Ui. If you use complaintList.getContent() then you are discarding pagination related details. which are valuable for setting up pagination data. Please change code for this

File: src/main/java/creativei/controller/ComplaintController.java [hunk 2]
```diff
@@ -27,30 +33,46 @@
...
+                status = customerService.saveExternalCustomer(complaintVo);
+            }
+            if(!status){
+                String message = "Unable to create external customer.";
+                int code = ExceptionType.GENERAL_ERROR.getCode();
+                return new ResponseEntity<>(ResponseObject.getResponse(message, code), HttpStatus.BAD_REQUEST);
+            }
             Complaint complaint = new Complaint(complaintVo);
             complaint = complaintService.save(complaint);
             return new ResponseEntity<>(ResponseObject.getResponse(complaint), HttpStatus.OK);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: Use try/catch. Where are you handling any exception that may occur on the service/dao layer?

File: src/main/java/creativei/controller/ComplaintController.java [hunk 3]
```diff
@@ -65,25 +87,71 @@ ResponseObject getAllComplaint(final Pageable pageable){
...
-            return ResponseObject.getResponse(ExceptionType.GENERAL_ERROR.getMessage()
-                    , ExceptionType.GENERAL_ERROR.getCode());
+            return new ResponseEntity<>(ResponseObject.getResponse(ExceptionType.GENERAL_ERROR.getMessage()
+                    , ExceptionType.GENERAL_ERROR.getCode()),HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
+    @GetMapping("/priority")
+    public @ResponseBody
+    ResponseObject getAllComplaintPriority() {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: Use ResponseEntity for every request on Complaint controller.

File: src/main/java/creativei/controller/ComplaintController.java [hunk 4]
```diff
@@ -65,25 +87,71 @@ ResponseObject getAllComplaint(final Pageable pageable){
...
+        List<String> complaintPriorityList = ComplaintPriority.ComplaintPriorityList;
+        return ResponseObject.getResponse(complaintPriorityList);
+    }
+
+    @GetMapping("/status")
+    public @ResponseBody
+    ResponseObject getAllComplaintStatus() {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: ResponseEntity

File: src/main/java/creativei/controller/ComplaintController.java [hunk 5]
```diff
@@ -65,25 +87,71 @@ ResponseObject getAllComplaint(final Pageable pageable){
...
+        List<String> complaintStatusList = ComplaintStatus.ComplaintStatusList;
+        return ResponseObject.getResponse(complaintStatusList);
+    }
+
+    @GetMapping("/type")
+    public @ResponseBody
+    ResponseObject getAllComplaintType() {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: ResponseEntity

File: src/main/java/creativei/controller/ComplaintController.java [hunk 6]
```diff
@@ -65,25 +87,71 @@ ResponseObject getAllComplaint(final Pageable pageable){
...
+        List<String> complaintTypeList = ComplaintType.ComplaintTypeList;
+        return ResponseObject.getResponse(complaintTypeList);
+    }
+
+    @GetMapping("/tatHours")
+    public @ResponseBody
+    ResponseObject getTatHours() {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: ResponseEntity

File: src/main/java/creativei/controller/ComplaintController.java [hunk 7]
```diff
@@ -65,25 +87,71 @@ ResponseObject getAllComplaint(final Pageable pageable){
...
+        Boolean status = true;
+        try{
+            Complaint complaint = new Complaint(complaintVo);
+            complaint = complaintService.save(complaint);
+            return new ResponseEntity<>(ResponseObject.getResponse(complaint), HttpStatus.OK);
+        }catch(Exception e) {
+            logger.error(e.getMessage(), e);
+            String message = ExceptionType.GENERAL_ERROR.getMessage();
+            int code = ExceptionType.GENERAL_ERROR.getCode();
+            return new ResponseEntity<>(ResponseObject.getResponse(message, code), HttpStatus.BAD_REQUEST);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/ComplaintController.java]: Use internal server error for HttpStatus

File: src/main/java/creativei/controller/MainController.java [hunk 8]
```diff
@@ -243,4 +250,32 @@ public void generateSendPdf(HttpServletResponse res, HttpServletRequest req) {
 		}
 	}
 
+	@RequestMapping(value = "/company" , method = RequestMethod.GET)
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: rename to /selfinfo

Comment (manojkrsaini09) [reply src/main/java/creativei/controller/MainController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 6 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 5 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 3 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 2 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: code wrapped in try, catch block.

Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: image

File: src/main/java/creativei/controller/ComplaintController.java (hunk 1 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: I have to explore pagination once to worked on it.

Comment (aayushKhandpur) [reply src/main/java/creativei/controller/ComplaintController.java]: Please explore. Pass the whole Page<Complaint> object to ui and then use content there.

**Comment by aayushKhandpur** [comment]
@manojkrsaini09 When can I expect an updated commit?

File: src/main/java/creativei/controller/ComplaintController.java (hunk 1 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 3 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 4 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 5 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 6 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 7 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: done

File: src/main/java/creativei/controller/MainController.java (hunk 8 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/MainController.java]: done

File: src/main/java/creativei/controller/ComplaintController.java (hunk 2 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/ComplaintController.java]: addressed

**Pull request merged**